validator-cli/src/utils/arbMsgExecutor.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ChildTransactionReceipt,
+  ArbitrumProvider,
+  ChildToParentMessageWriter,
+  ChildToParentMessageStatus,
+} from "@arbitrum/sdk";
+import { Wallet } from "@ethersproject/wallet";
+import { JsonRpcProvider, TransactionReceipt } from "@ethersproject/providers";
+import { Signer } from "@ethersproject/abstract-signer";
+import { ContractTransaction } from "@ethersproject/contracts";
+
+// Execute the child-to-parent (arbitrum-to-ethereum) message, for reference see: https://docs.arbitrum.io/sdk/reference/message/ChildToParentMessage
+async function messageExecutor(trnxHash: string, childRpc: string, parentRpc: string): Promise<ContractTransaction> {
+  const PRIVATE_KEY = process.env.PRIVATE_KEY;
+  const childJsonRpc = new JsonRpcProvider(childRpc);
+  const childProvider = new ArbitrumProvider(childJsonRpc);
+  const parentProvider = new JsonRpcProvider(parentRpc);
+
+  let childReceipt: TransactionReceipt | null;
+  try {
+    childReceipt = await childProvider.getTransactionReceipt(trnxHash);
+  } catch (error) {
```

Comment:
Avoid using `try/catch` for masquerading/renaming errors. 

- It makes the code more dense. Note that for a single line, you had to add six.
- It is tricky to have it done right. In this case, for example, the original stack trace and error name is lost.

Reply:
Understood 🫡